File: app/src/main/java/net/chris/demo/guardian/database/converter/DateTimeStructConverter.java
package net.chris.demo.guardian.database.converter;

import com.raizlabs.android.dbflow.converter.TypeConverter;

import net.chris.demo.guardian.system.DateTimeStruct;

/**
 * Converts {@link DateTimeStruct}
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
public final class DateTimeStructConverter extends TypeConverter<String, DateTimeStruct> {

    public DateTimeStructConverter() {
    }

    @Override
    public String getDBValue(DateTimeStruct model) {
        return model == null ? null : model.toString();
    }

    @Override
    public DateTimeStruct getModelValue(String data) {
        return data == null ? null : new DateTimeStruct(data);
    }
}

File: app/src/main/kotlin/net/chris/demo/guardian/database/converter/DateTimeStructConverter.kt
package net.chris.demo.guardian.database.converter

import com.raizlabs.android.dbflow.converter.TypeConverter

import net.chris.demo.guardian.system.DateTimeStruct

/**
 * Converts [DateTimeStruct]
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
class DateTimeStructConverter : TypeConverter<String, DateTimeStruct>() {

    override fun getDBValue(model: DateTimeStruct?): String? {
        return model?.toString()
    }

    override fun getModelValue(data: String?): DateTimeStruct? {
        return if (data == null) null else DateTimeStruct(data)
    }
}
